### PR TITLE
[Notifier] Add unit tests

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | 5.0
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | - 
| License       | MIT
| Doc PR        | - 

- [x] `AbstractChannel`
- [x] `ChannelPolicy`
- [x] `RecipientTest` (done in PR #35773)
- [x] `EmailRecipientTest` (done in PR #35773)
- [x] `SmsRecipientTest` (done in PR #35773)
- [x] `Transports` (see PR #35834)

### CODE REVIEW

File: src/Symfony/Component/Notifier/Tests/Channel/AbstractChannelTest.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Notifier/Tests/Channel/AbstractChannelTest.php]: we don't use strict type

Comment (nicolas-grekas) [reply src/Symfony/Component/Notifier/Tests/Channel/AbstractChannelTest.php]: the license header is missing

File: src/Symfony/Component/Notifier/Tests/Channel/AbstractChannelTest.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Notification\Notification;
+use Symfony\Component\Notifier\Recipient\Recipient;
+
+/**
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
+ */
+class AbstractChannelTest extends TestCase
+{
+    public function test_a_channel_cannot_be_constructed_without_transport_and_bus(): void
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Notifier/Tests/Channel/AbstractChannelTest.php]: we use camel-case
and `void` should be removed: it means nothing on test cases - they're pure noise

**Review by nicolas-grekas — COMMENTED** [review]
thanks for starting this
I suppose these don't apply on 5.0? if they do, the PR should target 5.0.

**Comment by jschaedl** [comment]
@nicolas-grekas 

> I suppose these don't apply on 5.0? if they do, the PR should target 5.0.

I would like to add tests for the `EmailChannel`, `SmsChannel` and `ChatChannel` here as well, but wanted to wait for #35773 first.

If you are fine with only merging `AbstractChannelTest` and `ChannelPolicyTest` we could indeed target 5.0 for now. Then I could do the tests for `EmailChannel`, `SmsChannel` and `ChatChannel` later or in #35773.

What do you think?

**Comment by nicolas-grekas** [comment]
Yes please: tests that run on 5.0 should be submitted there. Then tests for new notifiers should go on master.

**Comment by nicolas-grekas** [comment]
(failures unrelated, needs review)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jschaedl.

**Pull request merged**